applications/Unity.GrantManager/src/Unity.GrantManager.Web/Views/Shared/Components/HistoryWidget/HistoryWidgetViewComponent.cs

```diff
@@ -0,0 +1,146 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Volo.Abp.AspNetCore.Mvc.UI.Widgets;
+using Volo.Abp.AspNetCore.Mvc;
+using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Volo.Abp.AspNetCore.Mvc.UI.Bundling;
+using Volo.Abp.Auditing;
+using Volo.Abp.AuditLogging;
+using Unity.GrantManager.History;
+using System.Threading;
+using Unity.GrantManager.Applications;
+using Volo.Abp.Identity;
+
+namespace Unity.GrantManager.Web.Views.Shared.Components.HistoryWidget
+{
+    [Widget(
+        RefreshUrl = "Widgets/History/RefreshHistory",
+        ScriptTypes = [typeof(HistoryWidgetScriptBundleContributor)],
+        StyleTypes = [typeof(HistoryWidgetStyleBundleContributor)],
+        AutoInitialize = true)]
+    public class HistoryWidgetViewComponent : AbpViewComponent
+    {  
+        private readonly IAuditLogRepository _auditLogRepository;
+        private readonly IdentityUserAppService _identityUserAppService;
+        private readonly IApplicationStatusRepository _applicationStatusRepository;
+
+        public HistoryWidgetViewComponent(IAuditLogRepository auditLogRepository,
+                                          IdentityUserAppService identityUserAppService,
+                                          IApplicationStatusRepository applicationStatusRepository)
+        {
+            _identityUserAppService = identityUserAppService;
+            _applicationStatusRepository = applicationStatusRepository;
+            _auditLogRepository = auditLogRepository;
+        }
+
+        public async Task<IViewComponentResult> InvokeAsync(Guid applicationId)
+        {
+            string? entityId = applicationId.ToString();
+            Dictionary<string, string> applicationStatusDict = await GetApplicationStatusDict();
+
+            HistoryWidgetViewModel model = new()
+            {       
+                ApplicationStatusHistoryList = await GetHistoryList(entityId, "ApplicationStatusId", applicationStatusDict),
+            };                 
+
+            return View(model);
+        }
+
+        public async Task<List<HistoryDto>> GetHistoryList(string? entityId, 
+                                                           string filterPropertyName, 
+                                                           Dictionary<string, string>? lookupDictionary) {
+
+            List<HistoryDto> historyList = new List<HistoryDto>();
+            string? sorting = null;
+            int maxResultCount = 50;
+            int skipCount = 0;
+            DateTime? startTime = null;
+            DateTime? endTime = null;
+            bool includeDetails = true;
+            Guid? auditLogId = null;
+            EntityChangeType? changeType = null;
+            string? entityTypeFullName = null;
+            CancellationToken cancellationToken = default;
+
+            var entityChanges = await _auditLogRepository.GetEntityChangeListAsync(
```

Comment:
Reading from the log repository which might grow substantially it might be working disabling tracking for this read: https://abp.io/docs/latest/framework/architecture/domain-driven-design/repositories?_redirected=B8ABF606AA1BDF5C629883DF1061649A#enabling-disabling-the-change-tracking

Reply:
Ok, pulled this into a HistoryAppService with the 

        [DisableEntityChangeTracking]
        public async Task<List<HistoryDto>> GetHistoryList(string? entityId, 
                                                           string filterPropertyName, 
                                                           Dictionary<string, string>? lookupDictionary) {